plotting/sound.py

```diff
@@ -65,16 +65,91 @@ def plot(self):
+            if subset.max() >= soniclayerdepth_value:
+                criticaldepth_idx = (np.abs(subset - soniclayerdepth_value)).argmin()
+                criticaldepth = self.temperature_depths.data[0][int(criticaldepth_idx) + int(minpos[0])]
+                criticaldepth_value = subset[criticaldepth_idx]
+
+                # Perform linear interpolation to get more accurate depth
+                if (criticaldepth_value > soniclayerdepth_value):
+                # Must also consider the previous value 
+                    criticaldepth_sec_value = subset[criticaldepth_idx - 1]
+                    criticaldepth_sec = self.temperature_depths.data[0][int(criticaldepth_idx - 1) + int(minpos[0])]
+                    criticaldepth_true = criticaldepth_sec + (soniclayerdepth_value - criticaldepth_sec_value) * (criticaldepth - criticaldepth_sec) / (criticaldepth_value - criticaldepth_sec_value)
+
+                else:
+                # Must also consider the next value
+                    criticaldepth_sec_value = subset[criticaldepth_idx + 1]
+                    criticaldepth_sec = self.temperature_depths.data[0][int(criticaldepth_idx + 1) + int(minpos[0])]
+                    criticaldepth_true = criticaldepth + (soniclayerdepth_value - criticaldepth_value) * (criticaldepth_sec - criticaldepth) / (criticaldepth_sec_value - criticaldepth_value)
+
+                ax.hlines(y=criticaldepth_true,xmin=soniclayerdepth_value -12, xmax=soniclayerdepth_value + 3)
+                criticaldepth_true = float("{0:.2f}".format(criticaldepth_true))
+                plt.text(soniclayerdepth_value + 4, criticaldepth_true, "Critical Depth: " + str(criticaldepth_true) + 'm')
+
+                # ~~~~~~~~~~~~~~
+
+                # DEPTH EXCESS
+
+                # First we have to find the last depth index with data
+                max_depth = self.temperature_depths.data[0][ss.count() - 1] #- criticaldepth_true
+                depthexcess = max_depth - criticaldepth_true
+                depthexcess = float("{0:.2f}".format(depthexcess))
+                plt.text( soniclayerdepth_value - 20, (criticaldepth_true + (depthexcess / 2)), "Depth Excess: " + str(depthexcess) + 'm')
+                # ~~~~~~~~~~~~
+
+            minpos = self.temperature_depths.data[0][minpos][0]
+            maxpos = np.where(self.sspeed[0] == maxspeed)
+
+            #plt.axvline(x=minspeed, ymin=0.5, ymax=1)
+            #plt.axhline(y=minpos)
+        
         ax.set_xlim([
             np.amin(self.sspeed) - (maxspeed - minspeed) * 0.1,
             np.amax(self.sspeed) + (maxspeed - minspeed) * 0.1,
         ])
+
+        if self.query.get('annotate'):
+            # Sound Speed Minima
+            minspeed = float("{0:.2f}".format(minspeed))
+            plt.text(minspeed + 2, minpos, "Sound Channel Axis: " + str(minspeed) + 'm')
+            ax.scatter(minspeed, minpos, s=200, marker='_')
+            # ~~~~~~~~~~~~~~~~~
```

Comment:
Would adding an empty line be sufficient in order to add white space?

Reply:
I'm not entirely sure what you mean or what your referring to?

Reply:
Instead of `# ~~`, would a new line be sufficient to increase legibility?

Reply:
Personally I think it's more clear with a visible barrier.
If you want it gone though I can remove it? I don't really care either way.